app/views/admin/enterprises/form/_vouchers.html.haml

```diff
@@ -3,30 +3,30 @@
     = t('.add_new')
 %br
 
-- if @enterprise.vouchers.present?
+- if @enterprise.vouchers.with_deleted.present?
   %table
     %thead
       %tr
         %th= t('.voucher_code')
         %th= t('.rate')
+        %th= t('.active')
         /%th= t('.label')
         /%th= t('.purpose')
         /%th= t('.expiry')
         /%th= t('.use_limit')
         /%th= t('.customers')
         /%th= t('.net_value')
     %tbody
-      - @enterprise.vouchers.each do |voucher|
+      - @enterprise.vouchers.with_deleted.order(deleted_at: :desc, code: :asc).each do |voucher|
         %tr
           %td= voucher.code
           %td= voucher.display_value  
+          %td= f.check_box :voucher_ids, { :multiple => true }, voucher.id, nil
```

Comment:
Clever, this looks a bit funny but I can see how it would work. I think it would be worth a comment to explain it though.